### ISSUE TITLE
(Feature request) Is it possible to integrate the plantUML? 

### ISSUE THREAD

**Issue by zhang-stephen** [zadam/trilium#3246]
### Describe feature

the [plantUML](https://plantuml.com/en/) is more powerful than mermaid, and fortunately I found a plugin to integrate plantUML into CKEditor.

plugin doc: https://plantuml.com/en/ckeditordemo 
plugin: https://github.com/mojitogames/ckeditor_plantuml

but it's not under active development... 😞

**Comment by zadam** [comment]
No, that's not planned. The main reason why Mermaid was chosen as opposed to (otherwise more mature) PlantUML is that PlantUML is a Java app and thus difficult to integrate into the node.js/electron app.

**Comment by zhang-stephen** [comment]
oh... that's so upset...

**Comment by f-steff** [comment]
This situation may have changed with the Javascript version of the PlantUML editor: https://github.com/plantuml/plantuml.js

But support for Kroki and configurable links to external editors (of which many are freely available on the net, and can also be selfhosted) is even higher on my wishlist. :-) 

From https://kroki.io/ :
Kroki provides a unified API with support for BlockDiag (BlockDiag, SeqDiag, ActDiag, NwDiag, PacketDiag, RackDiag), BPMN, Bytefield, C4 (with PlantUML), D2, DBML, Ditaa, Erd, Excalidraw, GraphViz, Mermaid, Nomnoml, Pikchr, PlantUML, Structurizr, SvgBob, Symbolator, TikZ, UMLet, Vega, Vega-Lite, WaveDrom, WireViz...

**Comment by zadam** [comment]
The thing is - the choice has been made to use Mermaid.js and switching to something different is no longer possible (BC break), and at the same time I don't want to integrate multiple libraries doing roughly the same. 

 https://kroki.io/ seems interesting for some use cases, but not really for integrating into app like Trilium - it seems quite massive and likely not available as pure JS.

**Comment by f-steff** [comment]
Thank you for your answer. I fully understand your standpoint, when it comes to integrate everything. However, that was not my full intent.
Instead I would like to be able to easily use links to an kroki instance which renders an image that's displayed as an image in Trilium context.

**Comment by vleeuwenmenno** [comment]
Is there any way to use DBML in Trilium at the moment? Or plugin support, so I can add kroki.io myself because I'd really like support for this as I'm using this quite often with https://dbdiagram.io but prefer to have this available on my notes.

**Comment by rauenzi** [comment]
You can definitely create a widget/script to make it available on your own instance, sure.
